pkg/controller/operators/olm/operator.go

```diff
@@ -403,7 +404,31 @@ func (a *Operator) syncClusterServiceVersion(obj interface{}) (syncError error)
 		"phase":     clusterServiceVersion.Status.Phase,
 	})
 
-	operatorNamespace, ok := clusterServiceVersion.GetAnnotations()["olm.operatorNamespace"]
+	operatorGroup := a.operatorGroupForActiveCSV(logger, clusterServiceVersion)
+
+	// don't process CSVs that are not active in an OperatorGroup
+	if operatorGroup == nil {
+		opgroups, err := a.lister.OperatorsV1alpha2().OperatorGroupLister().OperatorGroups(clusterServiceVersion.GetNamespace()).List(labels.Everything())
+		if err != nil {
+			// TODO: write out error status
+			logger.Warn("csv created in namespace without operator group, will not be processed")
+		}
+		if len(opgroups) == 1 {
+			a.addOperatorGroupAnnotations(&clusterServiceVersion.ObjectMeta, opgroups[0])
+			_, err := a.client.OperatorsV1alpha1().ClusterServiceVersions(clusterServiceVersion.GetNamespace()).Update(clusterServiceVersion)
+			if err != nil {
+				logger.WithField("opgroup", opgroups[0].GetName()).Error("error adding operatorgroup annotation")
+			}
+			return
+		}
+		if len(opgroups) > 1 {
+			logger.Warn("csv created in namespace with multiple operatorgroups, can't pick one automatically")
```

Comment:
Do we ever want to write the status back to the CSV when there are operator-group issues?